apps/astarte_trigger_engine/lib/astarte_trigger_engine/health/queries.ex

```diff
@@ -19,17 +19,22 @@
 defmodule Astarte.TriggerEngine.Health.Queries do
   alias Astarte.TriggerEngine.Config
   alias Astarte.Core.CQLUtils
+  alias Astarte.TriggerEngine.Realms
+  alias Astarte.TriggerEngine.Repo
   require Logger
 
+  import Ecto.Query
+
   def get_astarte_health(consistency) do
-    query = """
-    SELECT COUNT(*)
-    FROM #{CQLUtils.realm_name_to_keyspace_name("astarte", Config.astarte_instance_id!())}.realms
-    """
-
-    with {:ok, %Xandra.Page{} = page} <-
-           Xandra.Cluster.execute(:xandra, query, %{}, consistency: consistency),
-         {:ok, _} <- Enum.fetch(page, 0) do
+    keyspace_name =
+      CQLUtils.realm_name_to_keyspace_name("astarte", Config.astarte_instance_id!())
+
+    query =
+      from r in Realms,
+        prefix: ^keyspace_name,
+        select: count(r.realm_name)
```

Comment:
nit: you don't need to specify the primary key

```suggestion
        select: count()
```

better yet, you can also use [`Repo.aggregate`](https://hexdocs.pm/ecto/Ecto.Repo.html#c:aggregate/3).